seatunnel-plugin-discovery/src/main/java/org/apache/seatunnel/plugin/discovery/AbstractPluginDiscovery.java

```diff
@@ -107,15 +107,25 @@ public T createPluginInstance(PluginIdentifier pluginIdentifier) {
             try {
                 // use current thread classloader to avoid different classloader load same class error.
                 this.addURLToClassLoader.accept(classLoader, pluginJarPath.get());
+                ClassLoader finalClassLoader = classLoader;
```

Comment:
This class only load jars about plugin, load extra jars is this pr #2639 had been done. Please update your codes and execute `mvn clean package` at your local mechine.

Reply:
BTW, do you change any codes in flink module? I can not see you changed any codes in it.

Reply:
> This class only load jars about plugin, load extra jars is this pr #2639 had been done. Please update your codes and execute `mvn clean package` at your local mechine.

I know that.and i just use this method.
![image](https://user-images.githubusercontent.com/48236177/190297261-7276600f-12e9-4215-9a99-b4cb49d898fa.png)

Reply:
![image](https://user-images.githubusercontent.com/48236177/190297790-2c81d799-a6a5-4558-ba96-6e206b6392c5.png)
step1 will use dirver class ,and i Transfer the position step1 to step2.it will report error again.

Reply:
@CalvinKirs @Hisoka-X @EricJoy2048 @hailin0 PTAL.

Reply:
It comes to changes to the core logic, I need to verify this PR so that it is currently pending.

Reply:
You can use `addURLToClassLoader.accept` to load third party jar to classloader, but not here. `AbstractPluginDiscovery` used for load connector jar. But it use logic like `addURLToClassLoader` to load connector jar into current classloader. If you also want use logic like `addURLToClassLoader`, I think you should create new class like `ThirdPartyJarLoadService`, use same parameter to receive `addURLToClassLoader`.
PS: `addURLToClassLoader` logic shouldn't repeat, please extract public code.

Reply:
Do it in `registerPlugin` method maybe better. At now. `registerPlugin` only register to flink environment, but not load it into client classpath. You can call `addURLToClassLoader ` logic in there. And call `registerPlugin` to load third party jar first, then call Source/SinkExecuteProcessor to load connector jar.

Reply:
> 

Aggre with you.

Reply:
> Do it in `registerPlugin` method maybe better. At now. `registerPlugin` only register to flink environment, but not load it into client classpath. You can call `addURLToClassLoader ` logic in there. And call `registerPlugin` to load third party jar first, then call Source/SinkExecuteProcessor to load connector jar.

Aggre with you.

Reply:
ACK